coral-component-steplist/src/scripts/StepList.js

```diff
@@ -206,6 +208,35 @@ class StepList extends BaseComponent(HTMLElement) {
     this._reflectAttribute('size', this._size);
   
     this.classList.toggle(`${CLASSNAME}--small`, this._size === size.SMALL);
+
+    if (!this.items.length) {
+      return;
+    }
+
+    // update aria-label for all children
+    const _syncItemLabelled = () => {
+      const steps = this.items.getAll();
+      const stepsCount = steps.length;
+
+      for (let i = 0; i < stepsCount; i++) {
+        const step = steps[i];
+        if (step._elements.label.textContent.trim() !== '') {
+          if (this.size === size.SMALL) {
+            step.setAttribute('labelled', step._elements.label.textContent);
+          }
+          else {
+            step.removeAttribute('labelled');
+          }
+        }
+      }
+    };
+
+    if (typeof this.items.last()._syncTabIndex === 'function') {
```

Comment:
You can store `this.items.last()` in a variable to avoid calling it twice if the first condition is false.